roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -5,7 +5,7 @@
 # As as a final word, I prefer to keep the partition check instead of running ceph-disk prepare with "failed_when: false"
 # I believe it's safer
 - name: check if the device is a partition or a disk
-  shell: "echo '{{ item }}' | egrep '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
+  shell: "echo '{{ item }}' | egrep '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}$'"
```

Comment:
Ben, thanks this. This is exactly the method I used to test all the previous regex :).
The last one looks good to me, so feel free to amend your commit and push force the last changes. No need for an extra PR.

Thanks @bengland2 !